cmd/local-artifact-mirror/main.go

```diff
@@ -6,26 +6,57 @@ import (
 	"os"
 	"os/signal"
 	"path"
+	"strings"
 	"syscall"
 
-	"github.com/urfave/cli/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
 func main() {
-	ctx, cancel := signal.NotifyContext(
-		context.Background(),
-		syscall.SIGINT,
-		syscall.SIGTERM,
-	)
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
 	defer cancel()
+
 	name := path.Base(os.Args[0])
-	var app = &cli.App{
-		Name:     name,
-		Usage:    "Run or pull data for the local artifact mirror",
-		Commands: []*cli.Command{serveCommand, pullCommand},
+
+	InitAndExecute(ctx, name)
+}
+
+func RootCmd(ctx context.Context, v *viper.Viper, name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Run or pull data for the local artifact mirror",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			v.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Help()
+			os.Exit(1)
+			return nil
+		},
 	}
-	if err := app.RunContext(ctx, os.Args); err != nil {
+
+	cobra.OnInitialize(func() {
+		initConfig(v)
+	})
+
+	cmd.AddCommand(ServeCmd(ctx, v))
+	cmd.AddCommand(PullCmd(ctx, v))
+
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+
+	return cmd
+}
+
+func InitAndExecute(ctx context.Context, name string) {
+	v := viper.GetViper()
+	if err := RootCmd(ctx, v, name).Execute(); err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
 }
+
+func initConfig(v *viper.Viper) {
+	v.SetEnvPrefix("REPLICATED")
```

Comment:
You are going to have to change these to REPLICATED_DATA_DIR and REPLICATED_PORT.

The port one feels a little overly generic but your call.

https://github.com/replicatedhq/embedded-cluster/blob/fbc6db8672e10f18d811e50d9be1df8c162ab2e8/pkg/cmd/install.go#L90-L91